dune/grid/CpGrid.hpp

```diff
@@ -564,6 +571,27 @@ namespace Dune
             return 0;
         }
 
+        /// \brief returns the number of boundary segments within the macro grid
+        unsigned int numBoundarySegments() const
```

Comment:
It could be a surprise that this in O(n) in the number of faces, but I'll not worry about that now. I assume Dune make no run-time complexity guarantees for methods?